tcms/testcases/forms.py

```diff
@@ -622,7 +622,9 @@ class CaseTagForm(forms.Form):
 
     def populate(self, case_ids=None):
         if case_ids:
-            self.fields['o_tag'].queryset = TestTag.objects.filter(
-                testcase__in=case_ids).order_by('name').distinct()
+            tag_ids = TestCaseTag.objects.filter(
+                case__in=case_ids
+            ).values_list('tag').distinct()
+            self.fields['o_tag'].queryset = TestTag.objects.filter(pk__in=tag_ids).order_by('name')
```

Comment:
Replace ``testcase__in`` with ``cases__in`` works. No need to query ``TestCaseTag`` separately.